src/crimson/osd/osd_operation_external_tracking.h

```diff
@@ -36,7 +36,6 @@ struct LttngBackend
     ClientRequest::PGPipeline::RecoverMissing::
       BlockingEvent::ExitBarrierEvent::Backend,
     ClientRequest::PGPipeline::CheckAlreadyCompleteGetObc::BlockingEvent::Backend,
-    ClientRequest::PGPipeline::GetOBC::BlockingEvent::Backend,
```

Comment:
check_already_complete_get_obc was not intended to replace get_obc.
Attaching relevant discussion: https://github.com/ceph/ceph/pull/57966#pullrequestreview-2193027226

ClientRequest is the only user to check for already_complete which had to be done exclusively: 
```
  /**
   * The previous stage of recover_missing is a concurrent phase.
   * Checking for already_complete requests must done exclusively.
   * Since get_obc is also an exclusive stage, we can merge both stages into
   * a single stage and avoid stage switching overhead.
   */
```

To resolve this we could either:
* Introduce a new exclusive stage named check_for_already_complete and only then go into get_obc stage
* Simply move get_obc earlier, before checking for already complete

The first option could result in stage switch overheads, while the second could be not clear to readers as to why we wait for already_complete within get_obc stage.

Instead the two, it was decided to introduce the new stage check_already_complete_get_obc in order to be more explicit as to what is the logical stage we're at.

The other users of get_obc aren't actually checking for already_complete so they remained with the existing get_obc.

Reply:
The other two users are SnapTrimObjSubEvent and InternalClientRequest.  While it's true the only ClientRequest actually checks for completion, prior to that change all three used the get_obc stage.  After that change, there were two different exclusive stages taken at that point in the pipeline.

Stages serve two main purposes at the moment.  First, they ensure ordering.  However, that part is really only relevant for ClientRequest -- InternalClientRequest and SnapTrim don't have an external client to preserve ordering for.  The second purpose is mutual exclusion/invariant enforcement.  The main example of this second one is that the process stage ensures that the log entries we construct are submitted in the correct order -- otherwise the log could commit out of order.

Arguably, the only pipeline stage InternalClientRequest and SnapTrim need to take is process for the above reason.  Omitting the others would, I think, be harmless as all it would do allow them to reorder relative to each other and ClientRequest prior to entering process, but that shouldn't matter.

If they are going to use the stages, however, they need to be the same ones as ClientRequest -- taking GetOBC doesn't make sense if ClientRequest doesn't.